### ISSUE TITLE
 Rewrite old-style-class super calls #320 break my code

### ISSUE THREAD

**Issue by Czaki** [asottile/pyupgrade#547]
after upgrade from #320 my code is broken with a message:

```
  >       res = super().asarray(*args, **kwargs)
  E       TypeError: super(type, obj): obj must be an instance or subtype of type
```

The error may be caused because this part of the code is patching the external library to get progress information.

https://github.com/4DNucleome/PartSeg/blob/5e487764a5a35f4300f43ba81ba72e971565e040/package/PartSegImage/tifffile_fixes.py#L10-L36

is there an option to block this upgrade for this line?

**Comment by asottile** [comment]
yeah if you do terrible things like monkeypatch then all bets are off

I would recommend assigning your wrapped function in the global scope so your code does not break
